server/tests/test_serializers.py

```diff
@@ -1,17 +1,1421 @@
+def mock_jsonize_context(
+    user=None, session=None, locale_id=None, localizers={}, module_id=None
+):
+    if module_id:
+        return JsonizeModuleContext(
+            user=user,
+            session=session,
+            locale_id=locale_id,
+            module_id=module_id,
+            localizers=localizers,
+        )
+    else:
+        return JsonizeContext(
+            user=user, session=session, locale_id=locale_id, localizers=localizers
+        )
+
+
+def _jsonize_clientside_module_return(args: Dict[str, Any]) -> Dict[str, Any]:
+    result = {
+        "id_name": None,
+        "name": None,
+        "category": None,
+        "description": "",
+        "deprecated": None,
+        "icon": "url",
+        "loads_data": False,
+        "uses_data": True,
+        "help_url": "",
+        "has_zen_mode": False,
+        "has_html_output": False,
+        "row_action_menu_entry_title": "",
+        "js_module": "",
+        "param_fields": [],
+    }
+    result.update(args)
+    return result
+
+
+class JsonizeClientsideModuleTest(unittest.TestCase):
+    def test_with_js(self):
+        module = Module(
+            ModuleSpec(
+                **{
+                    "id_name": "testme",
+                    "name": "Test Module",
+                    "category": "Clean",
+                    "parameters": [],
+                }
+            ),
+            "var js = 1; console.log(js)",
+        )
+        result = jsonize_clientside_module(
+            module,
+            mock_jsonize_context(
+                locale_id="el",
+                localizers={
+                    "module.testme": MessageLocalizer("el", Catalog(), Catalog())
+                },
+            ),
+        )
+        expected = _jsonize_clientside_module_return(
+            {
+                "id_name": "testme",
+                "name": "Test Module",
+                "category": "Clean",
+                "js_module": "var js = 1; console.log(js)",
+            }
+        )
+        self.assertDictEqual(result, expected)
+
+    def test_no_params_translate(self):
+        module = Module(
+            ModuleSpec(
+                **{
+                    "id_name": "testme",
+                    "name": "Test Module",
+                    "category": "Clean",
+                    "parameters": [],
+                }
+            ),
+            "",
+        )
+        catalog = Catalog()
+        catalog.add("_spec.name", string="Translated name")
+        default_catalog = Catalog()
+        default_catalog.add("_spec.name", string="Default translated name")
+        result = jsonize_clientside_module(
+            module,
+            mock_jsonize_context(
+                locale_id="el",
+                localizers={
+                    "module.testme": MessageLocalizer("el", catalog, default_catalog)
+                },
+            ),
+        )
+        expected = _jsonize_clientside_module_return(
```

Comment:
`self.assertEqual(result["name"], "Translated name")`

If we compare the whole dict, the reader can't tell what's under test. We should have maybe one test that checks the entire structure; other than that, we should test the things we're testing, not the whole dict.